src/main/resources/api.yaml

```diff
@@ -207,3 +229,18 @@ definitions:
     required:
       - enrollmentPeriod
       - durationPeriod
+  SprintCreationRequest:
```

Comment:
dto?

Reply:
to czym to sie rozni od login request

Reply:
Przy loginie teoretycznie nie tworzysz nic nowego w bazie (tylko token i id w cache) - u nas w rozwiązaniu opartym o Bieda Logowanie :tm: czasem tworzy się user

Tutaj to chyba wedle konwencji naszej/ogólnej powinno się dziać w `SprintDto` - tak jak przy projekcie

Reply:
wedlug mnie bez sensu jest wysylac sprintdto z niekompletnymi danymi jako request utworzenia

Reply:
Inne szczegóły typu "id ma być puste" się waliduje na froncie

Reply:
ale po co

Reply:
Dla mnie też w sumie było bez sensu najpierw, bo tak się nie robi rzeczy w innych sytuacjach, ale w restowych api chyba tak

Reply:
no dobra

Reply:
W sumie id nie musi być koniecznie puste, będzie po prostu zignorowane.
Pytanie czy nie może być przypadku że się required pogryzą

Reply:
"Pytanie czy nie może być przypadku że się required pogryzą" good point

Reply:
Jak się pogryzą, to będziemy myśleć

Reply:
I tak nie będziemy tworzyć osobnej klasy na każdy use case

Reply:
`enrollmentPeriod` i `durationPeriod` na pewno muszą być zawsze

Reply:
Nie wiem. Przecież walidacji i tak się nie robi w pełni na `required`, nie?

Reply:
Przy zamykaniu sprintu no trzeba podać datę zakończenia, ale i tak może być response z errorem - możesz nie być adminem tego projektu, daty mogą się nie zgadzać

Reply:
Zresztą przy tym tworzeniu też będą errory - właśnie na admina, niepoprawne daty - mamy ten exception na kolejność w `LocalDatePeriod`, ale `enrollmentPeriod` i `durationPeriod` też muszą się zgadzać - wszystkie być w przyszłości, w dobrej kolejności i tak dalej

Reply:
Przy zamykaniu jeszcze jest opcja, że sprint już jest zamknięty, a przecież nie można dwa razy. Albo nie jest jeszcze `FINISHED`

Reply:
Masaaa rzeczy